SemanticUI/Modules/Dropdown/Select.elm

```diff
@@ -0,0 +1,327 @@
+
+
+{-| Select variations. For internal use only.
+
+Similar to `Dropdown.Variation`
+
+-}
+type Variation msg
+    = Ordinary
+    | Button (Button.Config msg)
+    | Inline
+    | Selection { compact : Bool }
+
+
+variationClassList : Variation msg -> List ( String, Bool )
+variationClassList variation =
+    case variation of
+        Ordinary ->
+            []
+
+        Button _ ->
+            []
+
+        Inline ->
+            [ ( "inline", True ) ]
+
+        Selection { compact } ->
+            [ ( "compact", compact ), ( "selection", True ) ]
+
+
+{-| Most general configuration that applies any `Select`.
+
+It is recommended that you use `inline`, `button`, or `select` to construct this record.
+
+-}
+type alias Config msg option =
+    { variation : Variation msg
+    , drawerState : Drawer.State
+    , identifier : String
+    , onSelect : option -> msg
+    , onToggle : Drawer.State -> msg
+    , attributes : List (Attribute msg)
+    , toggleEvent : Toggle.Event
+    , disabled : Bool
+    , optionAttributes : option -> List (Attribute msg)
+    , labels : List (Html msg)
+    , caret : Bool
+    , fluid : Bool
+    , scrolling : Bool
+    }
+
+
+{-| A type that represents the dropdown select.
+
+Use `inline`, `button`, or `select` to construct this record and `toHtml` or `toCustomHtml` to render it.
+
+-}
+type Select msg option
+    = Select (Config msg option)
+
+
+{-| Any other custom `Attribute`s to add to the select. Custom attributes
+will be added before `elm-semantic-ui` attributes.
+
+Identical to `Dropdown.attributes`
+
+-}
+attributes : List (Attribute msg) -> Select msg option -> Select msg option
+attributes a (Select config) =
+    Select { config | attributes = a }
+
+
+{-| Set `toggleEvent` on any `Select`
+
+Identical to `Dropdown.toggleEvent`
+
+-}
+toggleEvent : Toggle.Event -> Select msg option -> Select msg option
+toggleEvent a (Select config) =
+    Select { config | toggleEvent = a }
+
+
+{-| Set `disabled` on any `Select`.
+
+Identical to `Dropdown.disabled`
+
+-}
+disabled : Bool -> Select msg option -> Select msg option
+disabled a (Select config) =
+    Select { config | disabled = a }
+
+
+{-| Set `caret` on a `Select`.
+
+Identical to `Dropdown.caret`
+
+-}
+caret : Bool -> Select msg option -> Select msg option
+caret a (Select config) =
+    Select { config | caret = a }
+
+
+{-| Set `fluid on a`Select\`.
+
+Identical to `Dropdown.fluid`
+
+-}
+fluid : Bool -> Select msg option -> Select msg option
+fluid a (Select config) =
+    Select { config | fluid = a }
+
+
+{-| Set `scrolling` on a `Select`.
+
+Identical to `Dropdown.scrolling`
+
+-}
+scrolling : Bool -> Select msg option -> Select msg option
+scrolling a (Select config) =
+    Select { config | scrolling = a }
+
+
+{-| Everything needed to build the `Html msg` representation of a particular select dropdown.
+
+  - toDropdown - converts a `<div>` or an `<a>` element into a SemanticUI dropdown
```

Comment:
`toDropdown` works with anything, not just `div` or `a`:

```suggestion
  - toDropdown - converts a HTML element into a SemanticUI dropdown. Often used with `div` or `a`.
```

Reply:
I agree with the change, but just to mention, this is the list of elements where I know for sure that the `dropdown` class will have consistent styling (e.g. with corrected margins on the caret and misc stuff like that). Not a big deal, but just to explain the reasoning behind the comment.

Reply:
The other reason for the comment is sort of as preparation for properly supporting other dropdown root elements. (I think `<input>` and possibly `<select>` can also be used, but I need to check again.)

Reply:
Oh I nearly forgot, but `selection` is actually the tricky one. It connects the dropdown toggle element to the drawer in a way that will only layout as intended if you've picked an appropriate root element. So I wanted to point people to `<div>` by default, and `<a>` if you're building nested menus...